packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -513,18 +521,36 @@ - (void)saveImageWithPickerInfo:(NSDictionary *)info
   NSString *savedPath = [FLTImagePickerPhotoAssetUtil saveImageWithPickerInfo:info
                                                                         image:image
                                                                  imageQuality:imageQuality];
-  [self handleSavedPath:savedPath];
+  NSMutableArray *mutableArray = [[NSMutableArray alloc] init];
+  [mutableArray addObject:savedPath];
+
+  [self handleSavedPathList:mutableArray];
 }
 
-- (void)handleSavedPath:(NSString *)path {
+/**
+ * Applies NSMutableArray on the FLutterResult.
+ *
+ * NSString must be returned by FlutterResult if the single image
+ * mode is active. It is checked by @c maxImagesAllowed and
+ * returns the first object of the @c pathlist.
+ *
+ * NSMutableArray must be returned by FlutterResult if the multi-image
+ * mode is active. After the @c pathlist count is checked then it returns
+ * the @c pathlist.
+ *
+ * @param @pathList that should be applied to FlutterResult.
+ */
+- (void)handleSavedPathList:(NSMutableArray *)pathList {
   if (!self.result) {
     return;
   }
-  if (path) {
-    self.result(path);
+  if (pathList && (self.maxImagesAllowed == 1)) {
+    self.result(pathList.firstObject);
```

Comment:
Actually, I created a [PR](https://github.com/flutter/plugins/pull/3989) for this. But after some extra checks, we realized it would be a breaking change. So I closed it. What do you think?